terraform/examples/offline.tf

```diff
@@ -0,0 +1,761 @@
+  }
+}
+
+# In AWS, (eu-central-1)
+provider "aws" {
+  region = "eu-central-1"
+}
+
+# Used for the in-VPC EC2 endpoint.
+data "aws_security_group" "default" {
+  name   = "default"
+  vpc_id = module.vpc.vpc_id
+}
+
+module "vpc" {
+  source = "github.com/terraform-aws-modules/terraform-aws-vpc"
+
+  name = "offline"
+
+  cidr = "172.17.0.0/20"
+
+  azs             = ["eu-central-1a", "eu-central-1b", "eu-central-1c"]
+  private_subnets = ["172.17.0.0/22", "172.17.4.0/22", "172.17.8.0/22"]
+  public_subnets  = ["172.17.12.0/24", "172.17.13.0/24", "172.17.14.0/24"]
+
+  enable_dns_hostnames = true
+  enable_dns_support   = true
+
+  enable_dhcp_options      = true
+  dhcp_options_domain_name = "offline.zinfra.io"
+#  dhcp_options_domain_name_servers = 
+  
+  # In case we run terraform from within the environment.
+  # VPC endpoint for DynamoDB
+  enable_dynamodb_endpoint = true
+
+  # In case we run terraform from within the environment.
+  # VPC Endpoint for EC2
+  enable_ec2_endpoint              = true
+  ec2_endpoint_private_dns_enabled = true
+  ec2_endpoint_security_group_ids  = [data.aws_security_group.default.id]
+
+  enable_nat_gateway = true
+  one_nat_gateway_per_az = false
+# Use this only in productionish environments.
+#  one_nat_gateway_per_az = true
+
+  tags = {
+    Owner       = "Backend Team"
+    Environment = "Offline"
+    TrustZone   = "Crash"
+  }
+  vpc_tags = {
+    Owner       = "Backend Team"
+    Name        = "vpc-offline"
+    TrustZone   = "Crash"
+  }
+}
+
+# A SSH key, used during golden image creation. destroyed at the end of the process.
+resource "aws_key_pair" "crash-nonprod-deployer-julia" {
+  key_name   = "crash-nonprod-deployer-julia"
+  public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDANkhpvNjsWoYRt5ji82+K3QNXLNdr4V+1LRzI0PXUiIxGLw80LSEXuCQVD3S2OWE5BuBzZtTTD4gy+/Mic5U7/2tePtMJYLkTV8vwvD9nx1YQPUSiIiUcCVX6gpw/YmyACvso+2M3ageEHJzuad+ZNsuYah0g5Y1NFABbRnPHl2zIfyCl0efu1mz5ucY5Kxe8oHEH/gUEuhUSgpZpANjExdu44Gujry4dfypJ3F4nZzJXMWtEmiTnEZCJ24wfdBUnBqk4If2yLczFFRECtT6t4AQU6/AyJ5OYsX5AWcKr67qmBG8pJrwHJF0M9tBJhDkFZb+6XrUW3p3Oj8XNs2ubfcJofJaeP3ZzLz7LADXf8fvRcSruD9k7STGORptB4MFkF9vI7Xp/3fN71kVQ6zw1Pwdifnfkd03SOjj2JxOsSuXqXsSECFqFo7XF2m6xCB5fywdmHoOGzXy17FGGxpTunoe6C7zcHu5tVCTy6LZ6eFAzYK/h8lRnhvSiHXxXoX0= demo@boxtop"
+}
+
+# Finding AMIs:
+# https://cloud-images.ubuntu.com/locator/ec2/
+
+data "aws_ami" "ubuntu18LTS-ARM64" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["ubuntu/images/hvm-ssd/ubuntu-bionic-18.04-arm64-server-*"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  owners = ["099720109477"] # Canonical
+  }
+
+data "aws_ami" "ubuntu18LTS-AMD64" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["ubuntu/images/hvm-ssd/ubuntu-bionic-18.04-amd64-server-*"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  owners = ["099720109477"] # Canonical
+  }
```

Comment:
This block (for finding AMIs) could be deleted, or placed into `terraform/modules/aws_ami`.